services/spar/src/Spar/Scim/User.hs

```diff
@@ -780,13 +778,7 @@ assertExternalIdInAllowedValues allowedValues errmsg tid veid = do
     lift $
       ST.runValidExternalIdBoth
         (\ma mb -> (&&) <$> ma <*> mb)
-        ( \uref ->
-            getUserIdByUref (Just tid) uref <&> \case
-              (Spar.App.GetUserFound uid) -> Just uid `elem` allowedValues
-              Spar.App.GetUserNotFound -> Nothing `elem` allowedValues
-              Spar.App.GetUserNoTeam -> False -- this is never allowed (and also hopefully impossible)
-              Spar.App.GetUserWrongTeam -> False -- this can happen, but it's violating all our assertions
-        )
+        (\uref -> getUserByUrefUnsafe uref <&> (`elem` allowedValues) . fmap userId)
```

Comment:
provide `getUserIdByUrefUnsafe` to avoid having to go to brig and throw away the result.  I'll do this in a separate PR.